src/layers/embeddings.jl

```diff
@@ -1,27 +1,35 @@
 """
-    PatchEmbedding(patch_size)
-    PatchEmbedding(patch_height, patch_width)
+    PatchEmbedding(imsize::NTuple{2} = (224, 224); inchannels = 3, patch_size = (16, 16), 
+                   embedplanes = 768, norm_layer = identity, flatten = true)
 
-Patch embedding layer used by many vision transformer-like models to split the input image into patches.
+Patch embedding layer used by many vision transformer-like models to split the input image into 
+patches.
+
+# Arguments:
+- `imsize`: the size of the input image
+- `inchannels`: the number of channels in the input image
+- `patch_size`: the size of the patches
+- `embedplanes`: the number of channels in the embedding
+- `norm_layer`: the normalization layer
+- `flatten`: whether to flatten the input after the embedding
 """
-struct PatchEmbedding
-  patch_height::Int
-  patch_width::Int
-end
+function PatchEmbedding(imsize::NTuple{2} = (224, 224); inchannels = 3, patch_size = (16, 16), 
+                        embedplanes = 768, norm_layer = identity, flatten = true)
```

Comment:
Where does 768 come from?

Reply:
This is just a default value based on the 224-sized input, that's used in pretty much every paper without much changes (or with programmatically inserted values in case there are any changes) so...I just went with that 😅

Reply:
So is it calculated from 224 somehow? If it's calculated, then we should use the calculation formula as the default, not the result.

Reply:
On searching a little more, I found that all papers seem to be exposing this as one of the fundamental hparams to tweak - case in point, ViTs:

<img width="585" alt="image" src="https://user-images.githubusercontent.com/74202102/156227866-3d9b372a-c990-43f8-9eb7-a113cbbdc4d6.png">

(`embedplanes` follows from the timm implementation because it makes sense as the output of the `PatchEmbedding` layer but could always change that name to something more appropriate in case it's not informative enough)